### PR TITLE
Add Fetching mail with UID or Sequence number Index set.

### PR DESCRIPTION
**Pull request by pitiphong-p**
In Cocoa ordered collection API. IndexSet is used for represented index of the data in ordered collection. So I think it would be great for add the fetch mail API that used IndexSet as the parameter.

### CODE REVIEW

**Comment by pitiphong-p** [comment]
http://nshipster.com/nsindexset

File: Source/CTCoreFolder.h [hunk 1]
```diff
@@ -121,6 +121,65 @@
 */
 - (NSArray *)messagesFromUID:(NSUInteger)startUID to:(NSUInteger)endUID withFetchAttributes:(CTFetchAttributes)attrs;
 
+<<<<<<< Updated upstream
```
Comment (jwilling) [inline Source/CTCoreFolder.h]: Looks like there is a merge conflict here and on line 182.

**Comment by pitiphong-p** [comment]
@jwilling Sorry for the conflicted code. Pushed the fixed.

File: Source/CTCoreFolder.h (hunk 1 above)
Comment (pitiphong-p) [inline Source/CTCoreFolder.h]: Sorry about that. Pushed the new code already.

**Comment by mronge** [comment]
This is great, thanks!

**Pull request merged**